File: python/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complements = {}
        i = 0

        for num in nums:
            complement = target - num

            if complement in complements:
                return [complements[complement], i]

            complements[num] = i
            i += 1


# test here
nums = [2, 7, 11, 15]
target = 9
print(Solution().twoSum(nums, target))

File: javascript/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    complements = new Map()

    for (let i = 0; i < nums.length; i++) {
        complement = target - nums[i];
        
        if (complements.has(complement)) {
            return [ complements.get(complement), i ];
        } // if

        complements.set(nums[i], i);
    } // for
};

// test here
console.log(twoSum([3, 2, 4], 6));
